daemon/events/events.go

```diff
@@ -0,0 +1,144 @@
+package events
+
+import (
+	"context"
+	"time"
+
+	"github.com/alibaba/pouch/apis/types"
+
+	goevents "github.com/docker/go-events"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// Events is pubsub channel for events generated by the engine.
+type Events struct {
+	broadcaster *goevents.Broadcaster
+}
+
+// NewEvents return a new Events instance
+func NewEvents() *Events {
+	return &Events{
+		broadcaster: goevents.NewBroadcaster(),
+	}
+}
+
+// Publish sends an event. The caller will be considered the initial
+// publisher of the event. This means the timestamp will be calculated
+// at this point and this method may read from the calling context.
+// TODO: do we need lock when use broadcaster???
```

Comment:
we don't need to lock the broadcaster, because broadcaster uses channel to receive the event. please remove this comment.

Reply:
you are right, i just record this to reminder me to check the lock.